lib/cell.rb

```diff
@@ -0,0 +1,49 @@
+class Cell
+
+  attr_reader :coordinate, :empty, :ship, :fired_upon
+
+  def initialize(coordinate)
+    @coordinate = coordinate
+    @empty = true
+    @ship = nil
+    @fired_upon = false
+  end
+
+  def empty?
+    @empty
+  end
+
+  def place_ship(ship)
+    @empty = false
+    @ship = ship
+  end
+
+  def fired_upon?
+    @fired_upon
+  end
+
+  def fire_upon
+    @fired_upon = true
+    if @ship == ship
+      ship.hit
+    end
+  end
+
+  def render(show = false)
```

Comment:
This `show` argument is weird and maybe it's application will make more sense in the next iteration. In the instructions it is an optional argument that, if true, reveals if there is a ship. I don't yet understand how it will actually be used.

Reply:
Definitely a wired one but I think I get it. Either way I like how you did it.